lib/Posts/LinkTranslation.php

```diff
@@ -160,34 +162,46 @@ public function get_possible_links( WP_Post $post, string $post_lang, int $page,
 			$search_filter = $wpdb->prepare( "AND P.post_title LIKE %s", '%' . $wpdb->esc_like( $search ) . '%' );
 		}
 
+		$instr = [];
+		foreach ( $languages as $lang ) {
+			$instr[] = "INSTR(locale_info, '{$lang}') != 0 as {$lang}";
+		}
+		$instr = implode( ', ', $instr );
+
 		// FIXME: if there are any posts with multiple ubb_sources (bug), it can cause the post to show up in the list multiple times.
 		$possible_sources = $wpdb->get_results(
 			$wpdb->prepare(
-				"SELECT SQL_CALC_FOUND_ROWS
-				MIN(A.post_id) as post_id,
-				GROUP_CONCAT( CONCAT(A.post_id, 0x1F, locale, 0x1F, A.post_title) SEPARATOR 0x1D ) as group_info
+				"SELECT SQL_CALC_FOUND_ROWS post_id, group_info
```

Comment:
It's resource heavy. I was thinking about indexes, when we first encountered issues, maybe we benefit from having an index or two added to this table.